infra-as-code/bicep/modules/vwanConnectivity/vwanConnectivity.bicep

```diff
@@ -0,0 +1,199 @@
+/*
+SUMMARY: Module to deploy the Virtual WAN network topology and its components as per the Azure Landing Zone conceptual architecture - https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/virtual-wan-network-topology. This module draws parity with the Enterprise Scale implementation https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/subscriptionTemplates/vwan-connectivity.json
+DESCRIPTION: The following Azure resources will be deployed in a single resource group specified in the deployment parameter:
+              Virtual WAN
+              Public IP
+            The following resources are optional and can be configured by parameters:
+              Virtual Hub
+              VPN Gateway
+              ExpressRoute Gateway
+              Azure Firewall
+AUTHOR/S: Fai Lai @faister
+VERSION: 1.0.0
+*/
+
+@description('Prefix value which will be prepended to all resource names. Default: alz')
+param parCompanyPrefix string = 'alz'
+
+@description('The IP address range in CIDR notation for the vWAN virtual Hub to use. Default: 10.100.0.0/23')
+param parVhubAddressPrefix string = '10.100.0.0/23'
+
+@description('Switch which allows Azure Firewall deployment to be disabled. Default: false')
+param parAzureFirewallEnabled bool = false
+
+@description('Azure Firewall Tier associated with the Firewall to deploy. Default: Standard ')
+@allowed([
+  'Standard'
+  'Premium'
+])
+param parAzureFirewallTier string = 'Standard'
+
+@description('Public IP Address SKU. Default: Standard')
+@allowed([
+  'Basic'
+  'Standard'
+])
+param parPublicIPSku string = 'Standard'
+
+@description('Tags you would like to be applied to all resources in this module. Default: empty array')
+param parTags object = {}
+
+@description('Switch which allows Virtual Hub. Default: true')
+param parVirtualHubEnabled bool = true
+
+@description('Switch which allows VPN Gateway. Default: false')
```

Comment:
Should we also add P2S gateway?

Reply:
nope, trying to keep this simple on par with the ESLZ vWAN connectivity ARM template